src/mito2/src/access_layer.rs

```diff
@@ -114,15 +115,21 @@ impl AccessLayer {
                         metadata: request.metadata,
                         source: request.source,
                         storage: request.storage,
-                        upload_path: path,
+                        upload_path: file_path,
+                        index_upload_path: index_file_path,
                         remote_store: self.object_store.clone(),
                     },
                     write_opts,
                 )
                 .await?
         } else {
             // Write cache is disabled.
-            let mut writer = ParquetWriter::new(path, request.metadata, self.object_store.clone());
+            let mut writer = ParquetWriter::new(
+                file_path,
+                index_file_path,
```

Comment:
It is possible to pass an optional index writer to the parquet writer later so it doesn't need to care about how to create an index writer or the path of the index.

Reply:
Let’s delete this placeholder field first and make the decision next PR.